scripts/add_brownfield.py

```diff
@@ -443,6 +447,84 @@ def update_dynamic_ptes_capacity(
     ].values
 
 
+def extract_conventional_thermals_tyndp(
+    tyndp_carrier_mapping: pd.DataFrame,
+    tyndp_conventional_carriers: list[str],
+    grp_convs: bool,
+    options: dict,
+) -> list[str]:
+    """
+    Extract list of TYNDP conventional thermal generation technologies.
+
+    Filters and optionally groups TYNDP conventional carriers, then returns
+    a list of technology names including optional hydrogen technologies.
+
+    Parameters
+    ----------
+    tyndp_carrier_mapping : pd.DataFrame
+        Full TYNDP carrier mapping.
+    tyndp_conventional_carriers : list[str]
+        List of TYNDP conventional carrier names to include.
+    grp_convs : bool
+        Whether to group conventional carriers by open_tyndp_type.
+    options : dict
+        Configuration options containing
+
+    Returns
+    -------
+    list[str]
+        List of conventional thermal technology names.
+    """
+    # Prepare conventional mapping
+    tyndp_conventional_mapping = prepare_tyndp_conventional_mapping(
+        carrier_mapping=tyndp_carrier_mapping,
+        conventional_carriers=tyndp_conventional_carriers,
+    )
+
+    # Optionally group TYNDP conventionals
+    if grp_convs:
+        tyndp_conventional_mapping = tyndp_conventional_mapping.groupby(
+            "open_tyndp_type"
+        ).first()
+
+    # Get list of conventional thermal technologies
+    _, tyndp_conventional_thermals = get_tyndp_conventional_thermals(
+        mapping=tyndp_conventional_mapping,
+        include_h2_fuel_cell=options["hydrogen_fuel_cell"],
+        include_h2_turbine=options["hydrogen_turbine"],
+    )
+
+    return tyndp_conventional_thermals
+
+
+def remove_tyndp_conventionals_p(
+    n_p: pypsa.Network,
+    tyndp_conventional_thermals: list[str],
+):
+    """
+    Remove TYNDP conventional capacities from previous planning horizon network
+    as existing conventional capacities are given as cumulative input.
+
+    n_p : pypsa.Network
+        The network with the updated parameters from the previous planning horizon.
+    tyndp_conventional_thermals : list[str]
+        List of TYNDP conventional thermal technologies to remove capacities for.
+
+    Returns
+    -------
+    None
+        This function updates the network in place and does not return a value.
+    """
+    logger.info(
```

Comment:
it is a bit unclear that we are still tracking cumulative installed capacities also from the previous years, better replace with

"logger.info(
    "Remove cumulative TYNDP conventional capacities from previous planning horizon "
    "and replace with cumulative capacities from new planning horizon."
)"